### PR TITLE
update executor options for solana client controller

### PR DESCRIPTION
**Pull request by magj2006**
update for #119 :
    add definition of Solana devnet & mainnet ChinaID
    set unordered options when sending message to Solana Chain
    set value when calling addWhitelistToken to add token on Solana Chain

## Summary by CodeRabbit

- **New Features**
	- Enhanced response handling for token transfers and reward operations, providing additional context for specific chain actions.
	- Introduced constants for Solana blockchain messaging operations.
	- Added support for interactions with Solana client chains in the testing framework.

- **Bug Fixes**
	- Improved interchain communication logic for better functionality.

- **Documentation**
	- Updated function signatures to reflect new response formats and conditions.

### CODE REVIEW

**Review by MaxMustermann2 — COMMENTED** [review]
Please make similar edits in `ExocoreGatewayMock.sol` and add unit tests to verify the event emission in `ExocoreGateway.t.sol`

**Review by adu-web3 — APPROVED** [review]

**Comment by magj2006** [comment]
> Please make similar edits in `ExocoreGatewayMock.sol` and add unit tests to verify the event emission in `ExocoreGateway.t.sol`

Added, please review again.

File: src/core/ExocoreGateway.sol [hunk 1]
```diff
@@ -354,7 +354,11 @@ contract ExocoreGateway is
         }
         emit LSTTransfer(isDeposit, success, bytes32(token), bytes32(staker), amount);
 
-        response = isDeposit ? bytes("") : abi.encodePacked(lzNonce, success);
+        if (srcChainId == SOLANA_DEVNET_CHAIN_ID || srcChainId == SOLANA_MAINNET_CHAIN_ID) {
```
Comment (bwhour) [inline src/core/ExocoreGateway.sol:357]: can use the following code to reduce duplicate code

> function _isSolana(uint32 chainId) internal pure returns (bool) {
    return chainId == SOLANA_DEVNET_CHAIN_ID || chainId == SOLANA_MAINNET_CHAIN_ID;
}

File: src/core/ExocoreGateway.sol [hunk 2]
```diff
@@ -527,14 +537,38 @@ contract ExocoreGateway is
...
+    /// @dev Builds options for interchain messages based on chain and action
+    /// @param srcChainId The source chain ID
+    /// @param act The action being performed
+    /// @return options The built options
+    function _buildOptions(uint32 srcChainId, Action act) private pure returns (bytes memory) {
+        bytes memory options = OptionsBuilder.newOptions();
+
+        bool isSolana = srcChainId == SOLANA_DEVNET_CHAIN_ID || srcChainId == SOLANA_MAINNET_CHAIN_ID;
+
+        if (!isSolana) {
```
Comment (bwhour) [inline src/core/ExocoreGateway.sol:558]: though this part code works， the following will be simpler to read.

>  uint128 value;
    if (!isSolana) {
        options = options.addExecutorOrderedExecutionOption();
        value = DESTINATION_MSG_VALUE;
    } else {
        value = act == Action.REQUEST_ADD_WHITELIST_TOKEN 
            ? SOLANA_WHITELIST_TOKEN_MSG_VALUE 
            : DESTINATION_MSG_VALUE;
    }

Comment (magj2006) [reply src/core/ExocoreGateway.sol]: added _isSolana(.) function, please review it again.

File: src/core/ExocoreGateway.sol (hunk 1 above)
Comment (magj2006) [reply src/core/ExocoreGateway.sol]: added _isSolana(.) function, please review it again.

File: src/core/ExocoreGateway.sol [hunk 3]
```diff
@@ -552,16 +552,14 @@ contract ExocoreGateway is
     /// @return options The built options
     function _buildOptions(uint32 srcChainId, Action act) private pure returns (bytes memory) {
         bytes memory options = OptionsBuilder.newOptions();
+        uint128 value = DESTINATION_MSG_VALUE;
 
         if (!_isSolana(srcChainId)) {
-            // currently, LZ does not support ordered execution for Solana
             options = options.addExecutorOrderedExecutionOption();
+        } else if (act == Action.REQUEST_ADD_WHITELIST_TOKEN) {
```
Comment (bwhour) [inline src/core/ExocoreGateway.sol:559]: awesome.

**Review by bwhour — APPROVED** [review]
looks good.

**Pull request merged**